### PR TITLE
Use the new registry endpoint in the annoucements to the kubernetes-dev mailing list

### PR DESCRIPTION
**Pull request by upodroid**
#### What type of PR is this?

/kind enhancement
/kind cleanup

#### What this PR does / why we need it:

registry.k8s.io has been active for some time now but we still use the old endpoint in our announcement emails. 

https://github.com/kubernetes/kubernetes/commit/97efa303b763449284f9829a86a0bd313f38c8e9

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/release/issues/2745

#### Does this PR introduce a user-facing change?

```release-note
our mailing list announcements now uses the new registry.
```

/cc @saschagrunert @puerco @BenTheElder

### CODE REVIEW

File: pkg/notes/document/document.go [hunk 1]
```diff
@@ -30,7 +30,6 @@ import (
 	"github.com/sirupsen/logrus"
 	"golang.org/x/text/cases"
 	"golang.org/x/text/language"
-	"sigs.k8s.io/promo-tools/v3/image"
```
Comment (upodroid) [inline pkg/notes/document/document.go:33]: I need to update the constant in that library. I'll do that if the PR looks good and revert the change to this file.

Comment (upodroid) [reply pkg/notes/document/document.go:33]: nvm. 

https://kubernetes.slack.com/archives/CCK68P2Q2/p1667850096184829

TL;DR we need to have multiple complex conversations before merging https://github.com/kubernetes-sigs/promo-tools/pull/669

**Comment by ameukam** [comment]
IMHO, we should merge this after the patch releases are done for November 2022. we still don't have the backports for k8s.gcr.io => registry.k8s.io merged in the release branches.

File: pkg/notes/document/document.go [hunk 2]
```diff
@@ -129,7 +128,7 @@ func fetchImageMetadata(dir, tag string) (*ImageMetadata, error) {
 	const linkBase = "https://console.cloud.google.com/gcr/images/k8s-artifacts-prod/us/"
 
 	for manifest, tempArchitectures := range manifests {
-		imageName := strings.TrimPrefix(manifest, image.ProdRegistry+"/")
+		imageName := strings.TrimPrefix(manifest, "registry.k8s.io/")
```
Comment (saschagrunert) [inline pkg/notes/document/document.go:131]: Can we make `"registry.k8s.io/"` a const somewhere here?

**Review by saschagrunert — APPROVED** [review]

**Comment by upodroid** [comment]
This is ready now.

**Comment by upodroid** [comment]
> Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.

First time I'm seeing this. Merge method is squash anyway so can we skip this?

**Comment by upodroid** [comment]
This can be merged now. I rebased the PR.

**Review by saschagrunert — APPROVED** [review]
